docs/manual/java/guide/production/AkkaDiscoveryIntegration.md

```diff
@@ -26,6 +26,51 @@ The Guice module [AkkaDiscoveryServiceLocatorModule](api/index.html?com/lightben
 
 In development, your Lagom application will keep using the Lagom's dev-mode [ServiceLocator](api/index.html?com/lightbend/lagom/javadsl/api/ServiceLocator.html).
 
-Next, you will need to choose one of the available Akka Discovery implementations and configure it in your `application.conf` file. Consult the [Akka Discovery](https://doc.akka.io/docs/akka/2.5/discovery/index.html) documentation for further instructions.
+By default, Lagom uses [Aggregate multiple discovery methods](https://doc.akka.io/docs/akka/2.5/discovery/index.html#discovery-method-aggregate-multiple-discovery-methods). The first discovery method is set to Configuration and the second is set to DNS. 
+So the static definition of service endpoins has a priority over DNS discovery.
+
+To statically configure service endpoints in your `application.conf` file consult the [Aggregate multiple discovery methods](https://doc.akka.io/docs/akka/2.5/discovery/index.html#discovery-method-aggregate-multiple-discovery-methods) documentation.
+
+To configure service discovery with DNS Lagom provides some configuration settings. The default settings are following:
+```hocon
+  lagom.akka.discovery {
+  
+    # When the service lookup regex fails, the defaults are used for the port and protocol.
+    defaults {
+  
+      # The default port name. Blank if no port name should be added by default.
+      port-name = http
+  
+      # The default port protocol. Blank if no port protocol should be added by default.
+      port-protocol = tcp
+  
+      # The default scheme to use in returned URIs if not defined in the port-name-scheme-mappings.
+      scheme = http
+    }
+  
+    # A mapping of service names to lookup information. Each mapping should define the following:
+    #
+    # - lookup - An alternative name for the service. This can be configured with a simple name or a SRV lookup, for exampe:
+    #   * my-service (simple name)
+    #   * my-service.default.svc.cluster.local (simple fully-qualified name)
+    #   * _http._tcp.my-service (SRV)
+    #   * _http._tcp.my-service.default.svc.cluster.local (fully-qualified SRV)
+    # - scheme - The scheme to return in the URI. If undefined, it will use the default scheme lagom.akka.discover.defaults.scheme.
+    #
+    # For example:
+    # service-name-mappings {
+    #   my-service-name {
+    #     lookup = my-service.default.svc.cluster.local
+    #     scheme = http
+    #   }
+    # }
+    service-name-mappings {
+    }
+  
+  
+    # The timeout for a successful lookup.
+    lookup-timeout = 5 seconds
+  }
+```
```

Comment:
Nice docs. Thanks @lynxpluto !

BTW, instead of copy/pasting contents of the `reference.conf` it's better to import the _real thing_ :TM: using this [snippet syntax](https://github.com/lagom/lagom/blame/master/docs/manual/java/gettingstarted/UnderstandHello.md#L13):

```
@[anchor-name](../../path/to/file/reference.conf)
```

and then adding [anchors](https://github.com/lagom/lagom/blob/master/docs/manual/java/gettingstarted/code/docs/javadsl/gettingstarted/helloservice/HelloService.java#L15) on `reference.conf`.

Reply:
done

Reply:
Hmm, the relative path seems to be wrong:

```

[error] Bad relative link ../../../../../akka-service-locator/core/src/main/resources/reference.conf at /home/travis/build/lagom/lagom/docs/manual/java/guide/production/AkkaDiscoveryIntegration.md:35
[error] @[lagom-akka-discovery-reference-conf](../../../../../akka-service-locator/core/src/main/resources/reference.conf)
[error]                                        ^
[error] Bad relative link ../../../../../akka-service-locator/core/src/main/resources/reference.conf at /home/travis/build/lagom/lagom/docs/manual/scala/guide/production/AkkaDiscoveryIntegration.md:23
[error] @[lagom-akka-discovery-reference-conf](../../../../../akka-service-locator/core/src/main/resources/reference.conf)

```

see https://travis-ci.com/lagom/lagom/jobs/202199954#L880

You can test this locally running `cd docs; sbt markdownValidateDocs`.

Reply:
:laughing: But when you'll open "File changed" tab and select "Display rich diff" for any affected .md file you'll see that link redirects correctly. May be markdown validator doesn't maintain some paths?

Reply:
Yeah, I suspect the problem is the build runs from a basepath such that relative paths for the build are different then for the MD editors.

Reply:
I've checked locally and I can't find what is wrong.

I've compared with other snippets and other pages and everything looks correct. 

@jroper any idea?

Reply:
They should work in both places. What happens when you run the build on this branch locally?

Reply:
I had tried to run validations locally and got a lot of errors about bad relative paths :(
```
./sbt markdownValidateDocs
[info] Loading settings for project global-plugins from idea.sbt ...
[info] Loading global plugins from /home/lynxpluto/.sbt/1.0/plugins
[info] Loading settings for project plugins from plugins.sbt ...
[info] Loading settings for project lagom-build from build.sbt,plugins.sbt ...
[info] Loading project definition from /home/lynxpluto/dev-projects/github/lagom/lagom/project
[info] Updating ProjectRef(uri("file:/home/lynxpluto/dev-projects/github/lagom/lagom/project/"), "lagom-build")...
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 1 Scala source to /home/lynxpluto/dev-projects/github/lagom/lagom/project/target/scala-2.12/sbt-1.0/classes ...
[info] Done compiling.
[info] Loading settings for project root from build.sbt,version.sbt ...
[info] Resolving key references (65094 settings) ...
[info] Loading project definition from /home/lynxpluto/dev-projects/github/lagom/lagom/docs/project
[info] Loading settings for project docs from build.sbt ...
[info] Loading settings for project lagom-build from build.sbt,plugins.sbt ...
[info] Loading project definition from /home/lynxpluto/dev-projects/github/lagom/lagom/project
[info] Updating ProjectRef(uri("file:/home/lynxpluto/dev-projects/github/lagom/lagom/project/"), "lagom-build")...
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 1 Scala source to /home/lynxpluto/dev-projects/github/lagom/lagom/project/target/scala-2.12/sbt-1.0/classes ...
[info] Done compiling.
[info] Loading settings for project root from build.sbt,version.sbt ...
[info] Resolving key references (65431 settings) ...
[info] Set current project to docs (in build file:/home/lynxpluto/dev-projects/github/lagom/lagom/docs/)
[info] Serving javadocs from /home/lynxpluto/dev-projects/github/lagom/lagom/target/javaunidoc and scaladocs from /home/lynxpluto/dev-projects/github/lagom/lagom/target/unidoc. Rerun unidoc in root project to refresh
[info] Validating java documentation...
[info] [pass] Duplicate markdown file name test
[info] [pass] Missing wiki links test
[info] [pass] Missing anchor test
[error] Bad relative link api/index.html?com/lightbend/lagom/javadsl/persistence/ReadSideProcessor.html at /home/lynxpluto/dev-projects/github/lagom/lagom/docs/manual/java/releases/Migration12.md:34
[error] * Modify your read side processor to implement [`ReadSideProcessor`](api/index.html?com/lightbend/lagom/javadsl/persistence/ReadSideProcessor.html) instead of [`CassandraReadSideProcessor`](api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraReadSideProcessor.html).
[error]                                                                                 ^
[error] Bad relative link api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraReadSideProcessor.html at /home/lynxpluto/dev-projects/github/lagom/lagom/docs/manual/java/releases/Migration12.md:34
[error] * Modify your read side processor to implement [`ReadSideProcessor`](api/index.html?com/lightbend/lagom/javadsl/persistence/ReadSideProcessor.html) instead of [`CassandraReadSideProcessor`](api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraReadSideProcessor.html).
[error]                                                                                                                                                                                                          ^
[error] Bad relative link api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraReadSide.html at /home/lynxpluto/dev-projects/github/lagom/lagom/docs/manual/java/releases/Migration12.md:35
[error] * Ensure your read side processor has both [`CassandraReadSide`](api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraReadSide.html) and [`CassandraSession`](api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraSession.html) injected into its constructor.
[error]                                                                             ^
[error] Bad relative link api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraSession.html at /home/lynxpluto/dev-projects/github/lagom/lagom/docs/manual/java/releases/Migration12.md:35
[error] * Ensure your read side processor has both [`CassandraReadSide`](api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraReadSide.html) and [`CassandraSession`](api/index.html?com/lightbend/lagom/javadsl/persistence/cassandra/CassandraSession.html) injected into its constructor.

```

Reply:
You'll need to build the API docs first by running `sbt unidoc` in the root directory: https://github.com/lagom/lagom/tree/master/docs#build-the-scala-javadocs

Reply:
I updated PR and made all of affected files look closer to `docs/manual/java/guide/services/ServiceClients.md`, `docs/manual/scala/guide/services/ServiceClients.md`, `service/core/client/src/main/resources/reference.conf`
I believe that nesting level for the link ref is correct. Both referenced `reference.conf`s are known by IDE and redirects are succeed.
Starting `./sbt unidoc` after changes at root dir and `./sbt markdownValidateDocs` in `docs` dir has no effect. Changed MDs still erroneous as it seen from log output.
```
$ ./sbt markdownValidateDocs
[info] Loading settings for project global-plugins from idea.sbt ...
[info] Loading global plugins from /home/lynxpluto/.sbt/1.0/plugins
[info] Loading settings for project plugins from plugins.sbt ...
[info] Loading settings for project lagom-build from build.sbt,plugins.sbt ...
[info] Loading project definition from /home/lynxpluto/dev-projects/github/lagom/lagom/project
[info] Updating ProjectRef(uri("file:/home/lynxpluto/dev-projects/github/lagom/lagom/project/"), "lagom-build")...
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Loading settings for project root from build.sbt,version.sbt ...
[info] Resolving key references (65094 settings) ...
[info] Loading project definition from /home/lynxpluto/dev-projects/github/lagom/lagom/docs/project
[info] Loading settings for project docs from build.sbt ...
[info] Loading settings for project lagom-build from build.sbt,plugins.sbt ...
[info] Loading project definition from /home/lynxpluto/dev-projects/github/lagom/lagom/project
[info] Updating ProjectRef(uri("file:/home/lynxpluto/dev-projects/github/lagom/lagom/project/"), "lagom-build")...
[info] Done updating.
[warn] There may be incompatibilities among your library dependencies; run 'evicted' to see detailed eviction warnings.
[info] Compiling 1 Scala source to /home/lynxpluto/dev-projects/github/lagom/lagom/project/target/scala-2.12/sbt-1.0/classes ...
[info] Done compiling.
[info] Loading settings for project root from build.sbt,version.sbt ...
[info] Resolving key references (65431 settings) ...
[info] Set current project to docs (in build file:/home/lynxpluto/dev-projects/github/lagom/lagom/docs/)
[info] Serving javadocs from /home/lynxpluto/dev-projects/github/lagom/lagom/target/javaunidoc and scaladocs from /home/lynxpluto/dev-projects/github/lagom/lagom/target/unidoc. Rerun unidoc in root project to refresh
[info] Validating java documentation...
[info] [pass] Duplicate markdown file name test
[info] [pass] Missing wiki links test
[info] [pass] Missing anchor test
[error] Bad relative link ../../../../../akka-service-locator/core/src/main/resources/reference.conf at /home/lynxpluto/dev-projects/github/lagom/lagom/docs/manual/java/guide/production/AkkaDiscoveryIntegration.md:35
[error] @[lagom-akka-discovery-reference-conf](../../../../../akka-service-locator/core/src/main/resources/reference.conf)
[error]                                        ^
[info] [fail] Relative link test (1 errors)
[info] [pass] Missing wiki resources test
[info] [pass] Missing source files test
[info] [pass] Missing source segments test
[info] [pass] Orphan pages test
[info] Validating scala documentation...
[info] [pass] Duplicate markdown file name test
[info] [pass] Missing wiki links test
[info] [pass] Missing anchor test
[error] Bad relative link ../../../../../akka-service-locator/core/src/main/resources/reference.conf at /home/lynxpluto/dev-projects/github/lagom/lagom/docs/manual/scala/guide/production/AkkaDiscoveryIntegration.md:23
[error] @[lagom-akka-discovery-reference-conf](../../../../../akka-service-locator/core/src/main/resources/reference.conf)
[error]                                        ^
[info] [fail] Relative link test (1 errors)
[info] [pass] Missing wiki resources test
[info] [pass] Missing source files test
[info] [pass] Missing source segments test
[info] [pass] Orphan pages test
[error] java.lang.RuntimeException: Documentation validation failed
```
Have no idea about what did I miss :)

Reply:
@lynxpluto described exactly the same steps I followed yesterday. I will need to look into this more carefully. Thanks @jroper and @TimMoore for jumping in.

Reply:
I just realised there must be an empty line before the anchor reference. _That_ was the error. I've tested it locally. @lynxpluto, can you do the honors?

Reply:
Just tested it locally. Now It works. Thanks. :+1:  :smile:

Reply:
@ignasi35 Incredible catch 😱 
I seem I saw somewhere the plan of migrating documentation to Antora. No? 😂

Reply:
@ihostage the documentation written in antora would not be lagom itself and only some edge guides. But it's still unclear.